addons/event_sale/models/account_move.py

```diff
@@ -10,7 +12,34 @@ class AccountMove(models.Model):
     def _invoice_paid_hook(self):
         """ When an invoice linked to a sales order selling registrations is
         paid confirm attendees. Attendees should indeed not be confirmed before
-        full payment. """
+        full payment. Post tickets on paid orders too, accessible from portal."""
         res = super(AccountMove, self)._invoice_paid_hook()
-        self.mapped('line_ids.sale_line_ids')._update_registrations(confirm=True, mark_as_paid=True)
+        sale_order_line_ids = self.mapped('line_ids.sale_line_ids')
+        sale_order_line_ids._update_registrations(confirm=True, mark_as_paid=True)
+
+        values = self.env['event.registration'].sudo()._read_group(
+            groupby=['event_id', 'sale_order_id'],
+            domain=[('sale_order_line_id', 'in', sale_order_line_ids.ids)],
+            aggregates=['id:array_agg'],
+        )
+
+        order_pdfs = defaultdict(list)
+        for event, order, registration_ids in values:
+            pdf = self.env['ir.actions.report'].sudo()._render_qweb_pdf(
+                'event.action_report_event_registration_full_page_ticket',
+                registration_ids,
+            )[0]
+            order_pdfs[order] += [(f'Tickets - {event.name}.pdf', pdf)]
+
+        for order, pdfs in order_pdfs.items():
+            if pdfs:
+                order.with_context(lang=order.partner_id.lang or self.env.user.lang).message_post(
```

Comment:
In that case maybe translate the body using the same language (will it be the case ? not sure, as the '_' is embed in the method parameters).